python/py_object.cpp

```diff
@@ -23,19 +36,12 @@ PyObject::PyObject(const py::object &object) {
     py::gil_scoped_acquire acquire;
     py::module copy = py::module::import("copy");
     py::object deepcopy = copy.attr("deepcopy");
-    m_object = deepcopy(object);
+    return {deepcopy(object)};
```

Comment:
Do we need shallow copies somewhere? Seems odd for force a deep copy.

Reply:
Well, `copy()` (C++) is what is bound to `__deepcopy__`. Naming may seem odd, but is more consistent when comparing to xarray with`array.copy(deep=True)` (`deep=True` is the default). Basically:

- `__copy__` is bound to copy constructor
- `__deepcopy__` is bound to `copy()`

Reply:
Ok. This makes sense also for consistency with other dtypes. Even with nested variables, we never copy the inner variable but not its contents.

Reply:
Hmm, while this makes sense, shouldn't the copy constructor (not the free function) make a (shallow) copy rather than copying `py::object`? I assume the latter is just increasing the reference count?